src/core/gltf2exporter/dracocompressor_p.cpp

```diff
@@ -0,0 +1,314 @@
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "dracocompressor_p.h"
+#include "gltf2exporter_p.h"
+#include <Qt3DRender/QGeometry>
+#include <Qt3DRender/QAttribute>
+#include <Qt3DRender/QBuffer>
+#include <draco/mesh/mesh.h>
+#include <draco/compression/encode.h>
+#include <draco/attributes/geometry_attribute.h>
+#include <draco/attributes/geometry_indices.h>
+#include <draco/io/mesh_io.h>
+#include <QVector>
+#include <QFile>
+#include <QLoggingCategory>
+#include <QDataStream>
+#include <fstream>
+#include <iterator>
+#include <iostream>
+
+QT_BEGIN_NAMESPACE
+Q_LOGGING_CATEGORY(dracocompressor, "DracoCompressor")
+namespace Kuesa::DracoCompression {
+
+CompressedMesh::~CompressedMesh() = default;
+
+namespace {
+draco::GeometryAttribute::Type attributeTypeFromName(const QString &attributeName)
+{
+    auto type = draco::GeometryAttribute::Type::INVALID;
+
+    if (attributeName == Qt3DRender::QAttribute::defaultPositionAttributeName())
+        type = draco::GeometryAttribute::Type::POSITION;
+    else if (attributeName == Qt3DRender::QAttribute::defaultNormalAttributeName())
+        type = draco::GeometryAttribute::Type::NORMAL;
+    else if (attributeName == Qt3DRender::QAttribute::defaultColorAttributeName())
+        type = draco::GeometryAttribute::Type::COLOR;
+    else if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinateAttributeName())
+        type = draco::GeometryAttribute::Type::TEX_COORD;
+    else if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinate1AttributeName())
+        type = draco::GeometryAttribute::Type::TEX_COORD;
+    else if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinate2AttributeName())
+        type = draco::GeometryAttribute::Type::TEX_COORD;
+    else if (attributeName == Qt3DRender::QAttribute::defaultTangentAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+    else if (attributeName == Qt3DRender::QAttribute::defaultJointWeightsAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+    else if (attributeName == Qt3DRender::QAttribute::defaultJointIndicesAttributeName())
+        type = draco::GeometryAttribute::Type::GENERIC;
+
+    return type;
+}
+
+template<typename T>
+struct attribute_type_trait {
+    static constexpr draco::DataType type = draco::DataType::DT_INVALID;
+};
+
+template<>
+struct attribute_type_trait<float> {
+    static constexpr draco::DataType type = draco::DataType::DT_FLOAT32;
+};
+
+template<>
+struct attribute_type_trait<char> {
+    static constexpr draco::DataType type = draco::DataType::DT_INT8;
```

Comment:
clang warns that `type` is unused (as well as DT_UINT8, DT_INT16, DT_INT32, DT_UINT32, DT_FLOAT64).